velox/functions/prestosql/tests/ComparisonsTest.cpp

```diff
@@ -97,6 +97,60 @@ TEST_F(ComparisonsTest, betweenDate) {
   }
 }
 
+TEST_F(ComparisonsTest, betweenDecimal) {
+  this->options_.parseDecimalAsDouble = false;
```

Comment:
Changing shared state prevents tests from running in parallel.

Reply:
Moved it to the constructor. But on debugging I see, ctor being invoked for each test. Not sure if this helps at all. Do you have any suggestion in setting this option in thread safe way?

When you say running in parallel, are the tests within ComparisionTests run in parallel or are you referring to running in parallel with tests in some other test file?

Reply:
Using the `void SetUp() override` function to instantiate class variables is the preferred pattern.
Each test is its own class object. So each of your prior approaches should work as well.
Reference: https://github.com/google/googletest/blob/main/docs/primer.md

Reply:
@karteekmurthys can you address this comment as well? Thanks.